src/app/services/db.service.ts

```diff
@@ -51,7 +50,66 @@ export class DB {
             this.clients.set(tenantID, client);
             return client;
         }
-
         throw new Error('Invalid tenant ID');
     }
+
+    // Dans modifications: 
+    async findOrCreateChatRoom(tenantID: string, fromUserID: number, toUserID: number): Promise<number> {
+        const client: PrismaClient = this.getClient(tenantID); 
+        let room = await client.chatRoom.findFirst({
+            where: {
+                members: {
+                    every: { 
+                        userId: { in: [fromUserID, toUserID] }
+                    }
+                }
+            },
+            include: { members: true }
+        });
+        if (!room) {
```

Comment:
FYI, this is susceptible to race conditions. Unfortunately, there isn't a particularly clean way to handle it in this case since you are tying it to having multiple foreign keys present (rather than, say, a unique constraint where you can optimistically try to create the record and then let it fail if it already exists). One way to resolve this would be to add an additional query after this, and >1 records are found and the one just created isnt the first, remove the record you just created and return the first one.

That said, if you wind up adjusting the behavior to not tie rooms to a conversation between two specific people, the issue is irrelevant.